File: benchmarks/pandas/bench_format_thousands.py
"""Benchmark: format thousands separator on 100k numbers"""
import json, time

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10
data = [i * 1234.56 for i in range(ROWS)]

for _ in range(WARMUP):
    [f"{v:,.2f}" for v in data]

start = time.perf_counter()
for _ in range(ITERATIONS):
    [f"{v:,.2f}" for v in data]
total = (time.perf_counter() - start) * 1000
print(json.dumps({"function": "format_thousands", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_format_thousands.ts
/**
 * Benchmark: formatThousands on 100k numbers
 */
import { formatThousands } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;
const data = Array.from({ length: ROWS }, (_, i) => i * 1234.56);

for (let i = 0; i < WARMUP; i++) data.map((v) => formatThousands(v));
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) data.map((v) => formatThousands(v));
const total = performance.now() - start;
console.log(
  JSON.stringify({
    function: "format_thousands",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
